include/tester/vtcp/VTCPSession.hpp

```diff
@@ -18,11 +18,11 @@ class VTCPSession : public TCPSession
     std::function<void()> startManualTest,
     std::function<bsoncxx::document::value()> getHistoryHeaders,
     std::function<bsoncxx::document::value(const std::string &)> getHistoryLog):
-        TCPSession(std::move(socket), id, onMessageReceived, onDisconnect),
-        _startAutoTest(startAutoTest),
-        _startManualTest(startManualTest),
-        _getHistoryHeaders(getHistoryHeaders),
-        _getHistoryLog(getHistoryLog)
+        TCPSession(std::move(socket), std::move(id), std::move(onMessageReceived), std::move(onDisconnect)),
```

Comment:
Same here.
- `id` is `std::size_t` - which is a primitive. you can look at the definition of it and you'll see the type declaration there.
- Absolutely no need to pass the function variables to `TCPSession` which should be a generic class in the first place.
Make it so only `VTCPSession` holds these members and calls them whenever you need to.

Reply:
Done